helper/tlsutil/config.go

```diff
@@ -385,27 +415,66 @@ func (c *Config) IncomingTLSConfig() (*tls.Config, error) {
 
 // ParseCiphers parses ciphersuites from the comma-separated string into
 // recognized slice
-func ParseCiphers(cipherStr string) ([]uint16, error) {
+func ParseCiphers(tlsConfig *config.TLSConfig) ([]uint16, error) {
 	suites := []uint16{}
 
-	cipherStr = strings.TrimSpace(cipherStr)
+	cipherStr := strings.TrimSpace(tlsConfig.TLSCipherSuites)
 
-	var ciphers []string
+	var parsedCiphers []string
 	if cipherStr == "" {
-		ciphers = defaultTLSCiphers
+		parsedCiphers = defaultTLSCiphers
 
 	} else {
-		ciphers = strings.Split(cipherStr, ",")
+		parsedCiphers = strings.Split(tlsConfig.TLSCipherSuites, ",")
 	}
-	for _, cipher := range ciphers {
+	for _, cipher := range parsedCiphers {
 		c, ok := supportedTLSCiphers[cipher]
 		if !ok {
 			return suites, fmt.Errorf("unsupported TLS cipher %q", cipher)
 		}
 		suites = append(suites, c)
 	}
 
-	return suites, nil
+	// Ensure that the specified cipher suite list is supported by the TLS
+	// Certificate signature algorithm. This is a check for user error, where a
+	// TLS certificate could support RSA but a user has configured a cipher suite
+	// list of ciphers where only ECDSA is supported.
```

Comment:
Can you specify how this would change the existing error messages? I'm not sure I follow how pulling this functionality into a separate function would enable using more detailed error messages.